src/Chain.php

```diff
@@ -23,23 +25,33 @@ class Chain
     /**
      * Execute all filters in the chain
      *
-     * @param bool $isSet
+     * @param bool $isNotEmpty
      * @param mixed $value
      * @param array|null $filterData
      * @return FilterResult
+     * @throws Exception
      */
-    public function filter($isSet, $value = null, $filterData = null)
+    public function filter($isNotEmpty, $value = null, $filterData = null)
     {
         /** @var FilterRule $rule */
         foreach ($this->rules as $rule) {
             $rule->setFilterData($filterData);
-            if ($isSet || $rule->allowedNotSet()) {
-                $value = $rule->filter($value);
-                $isSet = true;
+            if ($isNotEmpty || $rule->allowedNotSet()) {
+                $filterResult = $rule->filter($value);
+
+                if (!$filterResult instanceof FilterResult) {
+                    throw new Exception(
```

Comment:
Then [LogicException](http://www.php.net/manual/en/class.logicexception.php) sounds like the best option to me, agreed?